pkg/webui/console/components/events/utils/index.js

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { getDeviceId, getApplicationId } from '@ttn-lw/lib/selectors/id'
+
+import DefaultPreview from '../previews/default'
+import DefaultSyntheticEventPreview from '../previews/synthetic/default'
+import SyntheticErrorEventPreview from '../previews/synthetic/error'
+
+import { eventIconMap, dataTypeMap, applicationUpMessages } from './definitions'
+
+export const getEventId = event => `${event.time}-${event.name}`
```

Comment:
Just as a note: with https://github.com/TheThingsNetwork/lorawan-stack/pull/3211 we can use the `unique_id` field.

Reply:
Yes. I'll add as soon as this is merged into `v3.10`.